peps/pep-0817.rst

```diff
@@ -0,0 +1,2528 @@
+and feature values. When evaluating wheels to install, the installer
+must determine whether variant properties are compatible with the
+system, and perform `variant ordering`_ based on the priority of the
+compatible variant properties. This is done in addition to determining
+the compatibility and ordering through tags. The ordering by variant
+properties takes precedence over ordering by tags. Null variants
+(variants with no variant properties) are preferred over non-variant
+wheels with the same tags.
+
+Every variant namespace is governed by a variant provider. There are two
+kinds of variant providers: install-time providers and ahead-of-time
+(AoT) providers. Install-time providers require plugins that are queried
+while installing wheels to determine the set of supported properties and
+their preference order. For AoT providers, this data is static and
+embedded in the wheel; it can be either provided directly by the
+wheel maintainer or queried at wheel build time from an AoT plugin. Both
+kinds of plugins are usually implemented as Python packages which
+implement the `provider plugin API`_.
+
+Installers, as well as other tools that need to verify variant wheel
+compatibility and are run in security-sensitive context, must not
+install or run code from any untrusted package for variant resolution
+without explicit user opt-in. Provider packages should take measures to
+guard against supply chain attacks, for example by vendoring all
+dependencies. Pinning dependencies is discouraged to comply with
+:pep:`517`
+build process, as provider plugins may be needed at build time and
+dependencies pinned to different versions could prevent different
+plugins from being installed simultaneously in the same environment.
+
+It is recommended that these tools vendor, reimplement or lock the most
+commonly used plugins to specific specific wheels after verifying
+their trustworthiness, to enable the ability to securely install variant
+wheels out-of-the-box. To reduce the maintenance costs, repositories of
+such vetted plugins could be maintained collaboratively and shared
+between different tools.
+
+For plugins and their dependencies not in such a pre-approved list,
+a trust-on-first-use mechanism for every version is recommended. In
+interactive sessions, the tool can explicitly ask the user for approval.
+In non-interactive sessions, the approval can be given using
+command-line interface options. It is important that the user is
+informed of the risk before giving such an approval.
+
+For a consistent experience between tools, variant wheels should be
+supported by default. Tools may provide an option to only use
+non-variant wheels. For scenarios requiring more control, providers can
+be marked as optional and then must be explicitly enabled by the user
+for the wheels to become available for installation.
+
+
+Summary of changes
+-------------------
+
+The Wheel Variant PEP introduces four key components:
```

Comment:
Per previous comment, this is the sort of thing that could go into the introduction of _Rationale_